worldedit-core/src/main/java/com/sk89q/worldedit/LocalSession.java

```diff
@@ -559,34 +561,40 @@ public boolean toggleSuperPickAxe() {
      * @throws IncompleteRegionException thrown if a region is not fully selected
      */
     public BlockVector3 getPlacementPosition(Actor actor) throws IncompleteRegionException {
-        checkNotNull(actor);
-        if (!placeAtPos1) {
-            if (actor instanceof Locatable) {
-                return ((Locatable) actor).getBlockLocation().toVector().toBlockPoint();
-            } else {
-                throw new IncompleteRegionException();
-            }
-        }
+        return this.placement.getPlacementPosition(selector, actor);
+    }
+
+    public Placement getPlacement() {
+        return this.placement;
+    }
 
-        return selector.getPrimaryPosition();
+    public void setPlacement(Placement placement) {
+        this.placement = placement;
     }
 
+    @Deprecated
     public void setPlaceAtPos1(boolean placeAtPos1) {
-        this.placeAtPos1 = placeAtPos1;
+        this.placement = new Placement(placeAtPos1 ? PlacementType.POS1 : PlacementType.PLAYER, BlockVector3.ZERO);
     }
 
+    @Deprecated
     public boolean isPlaceAtPos1() {
-        return placeAtPos1;
+        return this.placement.getPlacementType() == PlacementType.POS1;
     }
 
     /**
-     * Toggle placement position.
+     * Toggle placement position between POS1 and PLAYER.
      *
-     * @return whether "place at position 1" is now enabled
+     * @return whether placement is now at position 1
      */
+    @Deprecated
```

Comment:
as above. also maybe mention that the default will be pos1 if it's outside the domain (i.e. between POS1 and PLAYER)

Reply:
Will be added in my next push, including an explanation for what happens if it was outside the previous domain.